firestore/src/common/futures.h

```diff
@@ -55,8 +55,10 @@ template <typename T>
 Future<T> FailedFuture() {
   static auto* future = new Future<T>(FailedFuture<T>(
       Error::kErrorFailedPrecondition,
-      "This instance is in an invalid state. This is because the underlying "
-      "Firestore instance has been destructed."));
+      "The object that issued this future is in an invalid state. This can be "
+      "because:\n- the object was default-constructed and never reassigned;\n"
+      "- the object was moved from;\n- the Firestore instance with which the "
+      "object was associated has been destroyed."));
```

Comment:
Is it common to have error messages like this in C++? I know that in Java a line break in an error message would be pretty annoying and make the overall message (which includes the stacktrace) less readable.